staging/src/k8s.io/apimachinery/pkg/api/meta/help.go

```diff
@@ -57,6 +57,12 @@ func IsListType(obj runtime.Object) bool {
 		ok = err == nil
 
 		// cache only the first 1024 types
+		isListCache.lock.RLock()
+		if len(isListCache.byType) >= 1024 {
```

Comment:
This looks correct. It's possible to avoid the two RUnlock calls by doing:

```
isListCache.lock.RLock()
size := len(isListCache.byType) 
isListCache.lock.RUnlock()
if size >= 1024 {
...
}
...
```

Reply:
Since the check of map length is dependent on the value of exists, I think the formation I used should be fine.